src/FormGroup.js

```diff
@@ -1,44 +1,107 @@
-import React from 'react';
 import classNames from 'classnames';
+import React from 'react';
+import deprecated from 'react-prop-types/lib/deprecated';
+
+import { Sizes } from './styleMaps';
+import bootstrapUtils, { bsClass, bsSizes } from './utils/bootstrapUtils';
+import ValidComponentChildren from './utils/ValidComponentChildren';
+
+const propTypes = {
+  /**
+   * Sets `id` on `<FormControl>` and `htmlFor` on `<FormGroup.Label>`.
+   */
+  controlId: React.PropTypes.string,
+  /**
+   * @private
+   */
+  standalone: deprecated(
+    React.PropTypes.bool,
+    'Use a `<FormControl>` or `<InputGroup>` directly.'
+  ),
+  validationState: React.PropTypes.oneOf(['success', 'warning', 'error']),
+  /**
+   * @private
+   */
+  bsStyle: deprecated(
```

Comment:
Yeah, exactly. The class is `.has-error` instead of `.form-group-error`, so it feels semantically different from e.g. `.btn-danger` or whatever. I'm okay with saying that validation state and styling are somewhat orthogonal.